backend/python/app/services/entity_service.py

```diff
@@ -0,0 +1,46 @@
+from ..models.entity import Entity
+from ..models import db
+
+
+def get_entities():
+    # Entity is a SQLAlchemy model, we can use convenient methods provided
+    # by SQLAlchemy like query.all() to query the data
+    return [result.to_dict() for result in Entity.query.all()]
+
+
+def get_entity(id):
+    # get queries by the primary key, which is id for the Entity table
+    entity = Entity.query.get(id)
+    if entity is None:
+        raise ValueError
```

Comment:
Not sure if `ValueError` is the most fitting error type, I think if `entity` is `None`, it would be because the `id` was not found. I'd have a look at other error types, or raise an `Exception` with a custom message